main.tf

```diff
@@ -0,0 +1,28 @@
+provider "aws" {
+  region = var.region
+}
+
+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical
+}
```

Comment:
@karineiji can you explain why we need this block?

Reply:
they are parameters of our EC2 hardware. it is our image ubuntu-focal-20.04-amd64-server and we use hard virtual machine.